plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java

```diff
@@ -562,14 +561,17 @@ private Map<String, List<Object>> getColumnValues(SchemaTableName tableName, Tab
         List<FakerColumnHandle> dictionaryColumns = distinctValues.entrySet().stream()
                 .filter(entry -> entry.getValue() <= MAX_DICTIONARY_SIZE)
                 .map(entry -> columnHandles.get(entry.getKey()))
-                .filter(Objects::nonNull)
+                .filter(column -> column != null && !isNotRangeType(column.type()))
```

Comment:
you can use two consecutive filters:

.filter(Objects::nonNull)
.filter(FakerMetadata::isRangeType) // reverse condition

Reply:
Done